lib/pbench/server/api/resources/__init__.py

```diff
@@ -663,6 +685,8 @@ def _validate_query_params(self, request: Request, schema: Schema) -> JSONOBJECT
                 if len(values) > 1 and schema[key].type == ParamType.LIST:
                     json[key] = values
                 else:
+                    if len(values) > 1:
+                        raise RepeatedQueryParam(key)
                     json[key] = values[0]
```

Comment:
It seems a shame to ask the length question twice...so how about:
```
                if len(values) > 1:
                    if schema[key].type == ParamType.LIST:
                        json[key] = values
                    else:
                        raise RepeatedQueryParam(key)
                else:
                    json[key] = values[0]
```
or, this, which is prettier:
```
                if len(values) == 1:
                    json[key] = values[0]
                elif schema[key].type == ParamType.LIST:
                    json[key] = values
                else:
                    raise RepeatedQueryParam(key)
```
but, unless there's a really good reason for avoiding a "list of one", this strikes me as best:
```
                if schema[key].type == ParamType.LIST:
                    json[key] = values
                elif len(values) == 1:
                    json[key] = values[0]
                else:
                    raise RepeatedQueryParam(key)
```

Reply:
There was actually a reason to avoid the "list of one", because that would break my comma-list-string logic. However, now that you've twisted my arm and forced me to generalize that, "list of one" should be fine and I do prefer the simplified code here. Maybe that's another reason to re-work this whole mess. (Sigh.)

Reply:
I note that you went with option #2 which avoids the list-of-one result....

Reply:
And once again you've got me second-guessing myself.

I thought "why create a list of one element when it's not meaningful" ... however it occurs to me after the fact that this isn't quite correct. Now that I've made the list-ization of a string conditional on defining a separator, I'm back to having a real difference; if the subclass calling this common function didn't choose to define a separator for a LIST-type query parameter, the non-list single entry will be rejected.

Right now it doesn't matter, because "metadata" is the only LIST type query parameter, and it has separators. However, if we had a LIST with no separator defined, it'd be important to return a single value as a List[]. So I think something like your 3rd option is necessary, after all.